src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/SourceTextExtensions.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    internal static class SourceTextExtensions
+    {
+        public static TextChangeRange GetEncompassingTextChangeRange(this SourceText newText, SourceText oldText)
```

Comment:
I feel like this could use a doc comment since the name isn't 100% clear about the function (as someone with only some context in the area).